### ISSUE TITLE
Hyperparameter setting for RLHF

### ISSUE THREAD

**Issue by xwwwwww** [CarperAI/trlx#303]
### 📚 The doc issue

Could you please clarify a more detailed hyperparameter setting for the experiments in this blog, for example, batch size?

https://wandb.ai/carperai/summarize_RLHF/reports/Implementing-RLHF-Learning-to-Summarize-with-trlX--VmlldzozMzAwODM2#fine-tune-with-ppo

### Suggest a potential alternative/fix

update the README / documentation

**Comment by reciprocated** [comment]
You can click through the report into the run itself https://wandb.ai/carperai/summarize_RLHF/runs/lv9es38t/overview
Hyperparameters are under the "Overview > Config" tab

**Comment by xwwwwww** [comment]
Thanks!

**Comment by congchan** [comment]
> 
Hi 
I am trying the example/summarize_rlhf, and found that batch size parameters is not under controlled, no matter how I modified the train:batch_size, or deepspeed configs train_micro_batch_size_per_gpu, or "gradient_accumulation_steps", the output log are always train_batch_size ............. 2 and train_micro_batch_size_per_gpu  2, "gradient_accumulation_steps": 1,

[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   optimizer_legacy_fusion ...... False
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   optimizer_name ............... None
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   optimizer_params ............. None
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   pipeline ..................... {'stages': 'auto', 'partition': 'best', 'seed_layers': False, 'activation_checkpoint_interval': 0}
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   pld_enabled .................. False
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   pld_params ................... False
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   prescale_gradients ........... False
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   scheduler_name ............... None
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   scheduler_params ............. None
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   sparse_attention ............. None
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   sparse_gradients_enabled ..... False
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   steps_per_print .............. inf
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   train_batch_size ............. 2
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   train_micro_batch_size_per_gpu  2
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   use_node_local_storage ....... False
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   wall_clock_breakdown ......... False
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   world_size ................... 1
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   zero_allow_untested_optimizer  True
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   zero_config .................. stage=2 contiguous_gradients=True reduce_scatter=True reduce_bucket_size=500,000,000 allgather_partitions=True allgather_bucket_size=500000000 overlap_comm=False load_from_fp32_weights=True elastic_checkpoint=False offload_param=DeepSpeedZeroOffloadParamConfig(device='cpu', nvme_path=None, buffer_count=5, buffer_size=100,000,000, max_in_cpu=1,000,000,000, pin_memory=False) offload_optimizer=DeepSpeedZeroOffloadOptimizerConfig(device='cpu', nvme_path=None, buffer_count=4, pin_memory=False, pipeline=False, pipeline_read=False, pipeline_write=False, fast_init=False) sub_group_size=1,000,000,000 cpu_offload_param=None cpu_offload_use_pin_memory=None cpu_offload=None prefetch_bucket_size=50,000,000 param_persistence_threshold=100,000 model_persistence_threshold=sys.maxsize max_live_parameters=1,000,000,000 max_reuse_distance=1,000,000,000 gather_16bit_weights_on_model_save=False stage3_gather_fp16_weights_on_model_save=False ignore_unused_parameters=True legacy_stage1=False round_robin_gradients=False
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   zero_enabled ................. True
[2023-03-03 17:14:21,660] [INFO] [config.py:1013:print]   zero_optimization_stage ...... 2
[2023-03-03 17:14:21,660] [INFO] [config.py:998:print_user_config]   json = {
    "train_micro_batch_size_per_gpu": 2,
    "gradient_accumulation_steps": 1,
    "fp16": {
        "enabled": true,
        "min_loss_scale": 0.5,
        "fp16_scale_tolerance": 0.25,
        "opt_level": "O2",
        "auto_cast": false
    },
    "zero_optimization": {
        "stage": 2,
        "offload_param": {
            "device": "cpu"
        },
        "offload_optimizer": {
            "device": "cpu"
        },
        "allgather_partitions": true,
        "allgather_bucket_size": 5.000000e+08,
        "contiguous_gradients": true
    },
    "steps_per_print": inf,
    "bf16": {
        "enabled": false
    },
    "zero_allow_untested_optimizer": true
}

**Comment by Bo396543018** [comment]
> You can click through the report into the run itself https://wandb.ai/carperai/summarize_RLHF/runs/lv9es38t/overview Hyperparameters are under the "Overview > Config" tab

Hi, I find the number ofstep in this log is 2.5k. But use default hyparameters, it needs to train 6.4k steps. 
When the model is running greater than 3k steps, the loss starts to rise.
I would like to know the difference between the log of the actual run and the default parameters. Do I need to stop earlier in the actual run, or do I need to set more reasonable default parameters.
